src/main/scala/edu/ie3/simona/model/participant/HpModel.scala

```diff
@@ -222,24 +230,32 @@ final case class HpModel(
       relevantData: HpRelevantData,
       isRunning: Boolean,
       demandWrapper: ThermalDemandWrapper,
+      currentThermalGridState: ThermalGridState,
   ): HpState = {
-    val lastStateStorageQDot = lastState.thermalGridState.storageState
+    val qDotLastHouseState = lastState.thermalGridState.houseState
       .map(_.qDot)
       .getOrElse(zeroKW)
+    val currentEnergyOfThermalStorage = currentThermalGridState.storageState
+      .map(_.storedEnergy)
+      .getOrElse(zeroKWh)
+    val pThermalOfStorage = thermalGrid.heatStorage
+      .map(_.getpThermalMax: squants.Power)
+      .getOrElse(zeroKW)
 
     val (newActivePowerHp, newThermalPowerHp, qDotIntoGrid) = {
       if (isRunning)
         (pRated, pThermal, pThermal)
-      else if (lastStateStorageQDot < zeroKW)
-        (zeroKW, zeroKW, lastStateStorageQDot * (-1))
+      // If the house has req. demand and storage isn't empty, we can heat the house from storage.
       else if (
-        lastStateStorageQDot == zeroKW && (demandWrapper.houseDemand.hasRequiredDemand || demandWrapper.heatStorageDemand.hasRequiredDemand)
+        currentEnergyOfThermalStorage > zeroKWh && demandWrapper.houseDemand.hasRequiredDemand
+      ) {
+        (zeroKW, zeroKW, pThermalOfStorage)
+        // Edge case when em controlled: If the house was heated last state by Hp and surplus energy is gone now,
```

Comment:
What's "surplus energy"?
Also, you mean "has not reached target temp yet", or did it actually just try at the last tick?

Reply:
Changed surplus to `and setPower is below turnOn condition now,`. Within lastState we heated the house and we're still below targetTemp, then we would continue heating from storage even if the new Em request to reduce electric power, right?